### PR TITLE
Implement Cloud SQL for Postgres Support and Sample App

### LINKED ISSUE #240: Feature: Postgresql Support
Just ran into an issue with Quarkus building a native image using Google Cloud Build. I found this repo, however it only looks like MySQL is currently supported. 

2021-10-25 14:17:22,513 WARN [io.agr.pool] (agroal-11) Datasource '<default>': The SocketFactory class provided com.google.cloud.sql.postgres.SocketFactory could not be instantiated.
Default
2021-10-25 10:17:22.545 EDT
2021-10-25 14:17:22,544 ERROR [io.qua.run.Application] (main) Failed to start application (with profile prod): java.lang.ClassNotFoundException: com.google.cloud.sql.postgres.SocketFactory

### PR DESCRIPTION
**Pull request by dzou**
This introduces support for Cloud SQL libraries for Postgres and an additional sample application.

Fixes #240.

### CODE REVIEW

**Comment by suztomo** [comment]
https://cloud.google.com/sql/docs/postgres/create-instance exists.

**Comment by dzou** [comment]
> https://cloud.google.com/sql/docs/postgres/create-instance exists.

Nice catch, fixed. Thanks.

**Review by suztomo — APPROVED** [review]

File: native-image-samples/native-image-samples-client-library/cloud-sql-postgres-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+The application runs some simple Cloud SQL queries to demonstrate compatibility.
+
+## Setup Instructions
+
+1. Follow the [GCP Project and Native Image Setup Instructions](../../README.md).
+
+2.  Cloud SQL database setup:
+
+    1. Follow [these instructions](https://cloud.google.com/sql/docs/mysql/create-instance) to create a Cloud SQL instance.
```
[the remark this answers is not in the document]
Comment (dzou) [reply native-image-samples/native-image-samples-client-library/cloud-sql-postgres-sample/README.md]: I think that was an older diff, should have been fixed in https://github.com/GoogleCloudPlatform/native-image-support-java/pull/250/commits/49570ec49ccadcc33afeefc703c2245101692a59.

[the remark this answers is not in the document]
Comment (suztomo) [reply native-image-samples/native-image-samples-client-library/cloud-sql-postgres-sample/README.md]: Right, that comment was old comment. Sorry for confusion.

**Pull request merged**